backend/src/relationships/relationships.service.spec.ts

```diff
@@ -25,9 +32,82 @@ describe('RelationshipsService', () => {
     }).compile();
 
     service = module.get<RelationshipsService>(RelationshipsService);
+    usersRepository = module.get<UserRepository>(UserRepository);
+    relationshipsRepository = module.get<Repository<Relationship>>(
+      getRepositoryToken(Relationship),
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  describe('create', () => {
+    it('should throw without valid sourceId', async () => {
+      async function tryCreate() {
+        return await service.create({
+          sourceId: 1,
+          targetId: 2,
+          type: RelationshipTypeEnum.FRIEND,
+        });
+      }
```

Comment:
You could put that in a `beforeAll` in this `describe` block and adapt your tests.

Reply:
So that there is no code duplication (I noticed the subtle differences, but you could either adapt your tests, either let `tryCreate` accept 2 arguments).

Reply:
Conversation resolved as we are to redo the `Relationship` table.